lib/Controller/PageController.php

```diff
@@ -152,12 +183,37 @@ public function duplicateSession(): Response {
 	#[NoCSRFRequired]
 	#[PublicPage]
 	#[BruteForceProtection(action: 'talkRoomToken')]
+	#[BruteForceProtection(action: 'callUser')]
 	#[UseSession]
 	public function index(string $token = '', string $callUser = '', string $password = ''): Response {
 		$bruteForceToken = $token;
 		$user = $this->userSession->getUser();
 		if (!$user instanceof IUser) {
-			return $this->guestEnterRoom($token, $password);
+			if ($token === '') {
+				$room = $this->createPrivateRoom($callUser);
+				if ($room === null) {
+					$response = new TemplateResponse('core', '404-profile', [], 'guest');
+					$response->throttle(['action' => 'callUser', 'callUser' => $callUser]);
+
+					return $response;
+				}
+
+				try {
+					$this->limiter->registerAnonRequest(
+						'create-anonymous-conversation',
+						5, // Five conversations
+						60 * 60, // Per hour
+						$this->request->getRemoteAddress(),
+					);
+				} catch(IRateLimitExceededException $exception) {
+					return new TooManyRequestsResponse();
+				}
+				
+				// FIXME: add rate limiting
```

Comment:
Hmm, we added rate-limiting above, so I assume this FIXME can be removed?

Also, looking at this again, shouldn't the rate-limiting block be moved above the call to `createPrivateRoom()`? It looks wrong to error out after creating the new room when the user has been rate-limited...

Reply:
In Nextcloud we keep rate limiting/brute forcing in the controller, so that the services can be reused by other mechanisms.

But yeah, the registration should happen before creating the room 😅 (I can take care)

Regarding the comment, yes the fixme can be removed.